### PR TITLE
Add error code explanation for E0477

### PR DESCRIPTION
**Pull request by ldm0**
Part of #61137

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0477.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+fn print_string<'a>(s: Mutex<MyString<'a>>) {
+
+    i_want_static_closure(move || {     // error: this closure has lifetime 'a
+                                        //        rather than 'static
+        println!("{}", s.lock().unwrap().data);
+    });
+}
+```
+
+In this example, the closure doesn't satisfy the `'static` lifetime constraint.
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0477.md]:
```suggestion
In this example, the closure does not satisfy the `'static` lifetime constraint.
```

File: src/librustc_error_codes/error_codes/E0477.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+To fix this kind of error, you need to double check lifetime of the type. Here,
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0477.md]:
```suggestion
To fix this error, you need to double check the lifetime of the type. Here,
```

**Review by Dylan-DPC-zz — CHANGES_REQUESTED** [review]

File: src/librustc_error_codes/error_codes/E0477.md (hunk 1 above)
Comment (ldm0) [reply src/librustc_error_codes/error_codes/E0477.md]: Thanks! Fixed.

File: src/librustc_error_codes/error_codes/E0477.md (hunk 2 above)
Comment (ldm0) [reply src/librustc_error_codes/error_codes/E0477.md]: Thanks! Fixed.

**Pull request merged**